src/SnowflakeTransformation.php

```diff
@@ -0,0 +1,148 @@
+use Retry\Policy\SimpleRetryPolicy;
+use Retry\RetryProxy;
+
+class SnowflakeTransformation
+{
+    /** @var Connection $connection */
+    private $connection;
+
+    /** @var LoggerInterface $logger */
+    private $logger;
+
+    /** @var array $databaseConfig */
+    private $databaseConfig;
+
+    public function __construct(Config $config, LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+        $this->databaseConfig = $config->getDatabaseConfig();
+        $this->connection = $this->createConnection();
+    }
+
+    public function setSession(Config $config): void
+    {
+        $sessionVariables = [];
+        $sessionVariables['QUERY_TAG'] = sprintf("'%s'", json_encode(['runId' => getenv('KBC_RUNID')]));
+
+        if ($config->getQueryTimeout()) {
+            $sessionVariables['STATEMENT_TIMEOUT_IN_SECONDS'] = $config->getQueryTimeout();
+        }
+
+        array_walk($sessionVariables, function (&$item, $key): void {
+            $item = vsprintf(
+                '%s=%s',
+                [
+                    $key,
+                    $item,
+                ]
+            );
+        });
+
+        $query = sprintf('ALTER SESSION SET %s;', implode(',', $sessionVariables));
+        $this->runRetriableQuery($query, 'alter session');
+    }
+
+    public function processSteps(array $steps): void
+    {
+        foreach ($steps as $step) {
+            if ($step['execution'] === 'parallel') {
+                foreach ($step['blocks'] as $block) {
+                    $this->logger->info(sprintf('Start process block "%s"', $block['name']));
+                    $this->executionQueries($block['name'], $block['script']);
+                }
+            } else {
+                foreach ($step['blocks'] as $block) {
+                    $this->logger->info(sprintf('Start process block "%s"', $block['name']));
+                    $this->executionQueries($block['name'], $block['script']);
+                }
+            }
+        }
+    }
+
+    public function executionQueries(string $name, array $queries): void
+    {
+        foreach ($queries as $query) {
+            $this->logger->info(sprintf('Run query: %s', $query));
+            $this->runRetriableQuery($query, $name);
+        }
+    }
+
+    private function createConnection(): Connection
+    {
+        $connection = new Connection($this->databaseConfig);
+        try {
+            if (!empty($this->databaseConfig['schema'])) {
+                $connection->query(
+                    sprintf(
+                        'USE SCHEMA %s',
+                        QueryBuilder::quoteIdentifier($this->databaseConfig['schema'])
+                    )
+                );
+            }
+        } catch (CannotAccessObjectException $e) {
+            throw new UserException($e->getMessage(), 0, $e);
+        }
+
+        return $connection;
+    }
+
+    private function runRetriableQuery(string $query, string $errorMessage): void
```

Comment:
```suggestion
    private function runRetryableQuery(string $query, string $errorMessage): void
```
https://english.stackexchange.com/a/305274